File: 2023/java/src/test/java/org/frieder/aoc/day4/TestDay4B.java
package org.frieder.aoc.day4;

import org.frieder.aoc.day4.b.Solution4B;
import org.junit.Test;

import java.io.IOException;

import static org.junit.Assert.assertEquals;

public class TestDay4B {
    @Test
    public void testExample() throws IOException {
        assertEquals(30, Solution4B.getResult("../resources/4/test_b.txt"));
    }

    @Test
    public void testResult() throws IOException {
        assertEquals(5920640, Solution4B.getResult("../resources/4/input.txt"));
    }
}

File: 2023/php/test/TestDay4B.php
<?php declare(strict_types=1);

$loader = require __DIR__ . '/../vendor/autoload.php';

use day4\b\Solution4B;
use PHPUnit\Framework\TestCase;
use utils\TestLogger;

final class TestDay4B extends TestCase
{
    public function testExample(): void
    {
        $this->assertSame(30, Solution4B::getResult("../resources/4/test_b.txt", TestLogger::createTestLogger()));
    }

    public function testResult(): void
    {
        $this->assertSame(5920640, Solution4B::getResult("../resources/4/input.txt", TestLogger::createTestLogger()));
    }

}